src/main/java/com/zendesk/maxwell/MaxwellParser.java

```diff
@@ -187,10 +187,17 @@ private MaxwellAbstractRowsEvent processRowsEvent(AbstractRowEvent e) throws Sch
 					break;
 				case MySQLConstants.QUERY_EVENT:
 					QueryEvent qe = (QueryEvent) v4Event;
-					// TODO: need to handle this case.
-					// The MySQL guys say some storage engines will output a "COMMIT" QUERY_EVENT at the
-					// end of the stream.
-					throw new RuntimeException("Unhandled QueryEvent: " + qe);
+
+					if ( qe.getSql().toString().equals("COMMIT") ) {
+						// some storage engines(?) will output a "COMMIT" QUERY_EVENT instead of a XID_EVENT.
+						// not sure exactly how to trigger this.
+						if ( !list.isEmpty() )
+							list.getLast().setTXCommit();
+
+						return list;
```

Comment:
hmm, is a very interesting idea, but how would we handle it if maxwell was stopped in the middle of a transaction?  I guess we'd have to store the position of the last started transaction as well as the last row output??